packages/api/lib/rulesHelpers.js

```diff
@@ -579,10 +581,12 @@ function recordIsValid(rule) {
     error.message += 'Rule workflow is undefined.';
     throw error;
   }
-  if (!rule.rule.type) {
+  if (!rule.rule || !rule.rule.type) {
     error.message += 'Rule type is undefined.';
     throw error;
   }
+
+  recordIsValid(omitDeepBy(rule, isNull), ruleSchema, false);
```

Comment:
added schema validate.
omitDeepBy is used to removed nested 'null' value, such as:
`{rule: {type: 'onetime', value: null}}`